### PR TITLE
#831: [Disk Manager] log blocks count on WriteBlocks request

### PR DESCRIPTION
**Pull request by BarkovBG**
https://github.com/ydb-platform/nbs/issues/831

### CODE REVIEW

File: cloud/disk_manager/internal/pkg/clients/nbs/session.go [hunk 1]
```diff
@@ -247,10 +247,24 @@ func (s *Session) Write(
...
+
+	blocks := make([][]byte, blocksCount)
+	for blockInd := uint32(0); blockInd < blocksCount; blockInd++ {
+		startOffset := blockInd * blockSize
+		endOffset := (blockInd + 1) * blockSize
+		if endOffset > uint32(len(data)) {
+			endOffset = uint32(len(data))
+		}
+
+		blocks[blockInd] = data[startOffset:endOffset]
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/session.go:261]: Unnecessary copy

File: cloud/disk_manager/internal/pkg/clients/nbs/session.go [hunk 2]
```diff
@@ -247,10 +247,24 @@ func (s *Session) Write(
 
 	defer s.metrics.StatRequest("Write")(&err)
 
+	blockSize := s.volume.BlockSize
+	blocksCount := (uint32(len(data)) + blockSize - 1) / blockSize
+
+	blocks := make([][]byte, blocksCount)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/session.go:253]: Unnecessary load for garbage collector

File: cloud/disk_manager/internal/pkg/clients/nbs/session.go [hunk 3]
```diff
@@ -245,6 +245,15 @@ func (s *Session) Write(
 		return errors.NewRetriableErrorf("last rediscover was failed")
 	}
 
+	blockSize := s.volume.BlockSize
+	blocksCount := (uint32(len(data)) + blockSize - 1) / blockSize
+	logging.Debug(
+		ctx,
+		"WriteBlocks offset %v, count %v",
```
[the remark this answers is not in the document]
Comment (SvartMetal) [reply cloud/disk_manager/internal/pkg/clients/nbs/session.go:252]: Lets log disk id also?

File: cloud/disk_manager/internal/pkg/clients/nbs/session.go [hunk 4]
```diff
@@ -245,6 +245,15 @@ func (s *Session) Write(
 		return errors.NewRetriableErrorf("last rediscover was failed")
 	}
 
+	blockSize := s.volume.BlockSize
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/session.go:248]: s.BlockSize()

**Review by SvartMetal — APPROVED** [review]

**Review by SvartMetal — APPROVED** [review]

**Review by drbasic — APPROVED** [review]

**Pull request merged**